dexmaker/src/main/java/com/google/dexmaker/DexMaker.java

```diff
@@ -323,6 +327,50 @@ public void declare(FieldId<?, ?> fieldId, int flags, Object staticValue) {
         }
     }
 
+    // Generate a file name for the jar by taking a checksum of MethodIds and
+    // parent class types.
+    private String generateFileName() {
+        int checksum = 1;
+
+        Set<TypeId<?>> typesKeySet = types.keySet();
+        Iterator<TypeId<?>> it = typesKeySet.iterator();
+        int[] checksums = new int[typesKeySet.size()];
+        int i = 0;
+
+        while (it.hasNext()) {
+            TypeId<?> typeId = it.next();
+            TypeDeclaration decl = getTypeDeclaration(typeId);
+            Set<MethodId> methodSet = decl.methods.keySet();
+            checksums[i++] = 31 * decl.supertype.hashCode() + methodSet.hashCode();
```

Comment:
That's correct. If someone is using ProxyBuilder, the iterator basically points to a TypeId key set of size 1, and the methods key set contains all the methods that should be proxied (all the way up the chain).

I did not think it was necessary to include field declarations.